Integration/AndroidAAR/BabylonNative/src/main/cpp/BabylonNativeJNI.cpp

```diff
@@ -0,0 +1,170 @@
+    {
+        std::string filenameStr{filename};
+        if (filenameStr.substr(0, strlen(Root)) == std::string(Root))
+        {
+            filename += strlen(Root) + 1; // + "/"
+        }
+        std::vector<char> buffer;
+        AAsset *asset = AAssetManager_open(g_assetMgrNative, filename,
+                                           AASSET_MODE_UNKNOWN);
+        if (asset != nullptr)
+        {
+            size_t size = AAsset_getLength64(asset);
+            buffer.resize(size);
+            AAsset_read(asset, buffer.data(), size);
+            AAsset_close(asset);
+        }
+        return buffer;
+    }
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_initEngine(JNIEnv* env, jobject obj,
+                                                   jobject assetMgr)
+{
+    auto asset_manager = AAssetManager_fromJava(env, assetMgr);
+    g_assetMgrNative = asset_manager;
+    jboolean iscopy;
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_finishEngine(JNIEnv* env, jobject obj)
+{
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_surfaceCreated(JNIEnv* env, jobject obj, jobject surface)
+{
+    if (!runtime)
+    {
+        ANativeWindow *window = ANativeWindow_fromSurface(env, surface);
+
+        int32_t width  = ANativeWindow_getWidth(window);
+        int32_t height = ANativeWindow_getHeight(window);
+
+        runtime = std::make_unique<Babylon::RuntimeAndroid>(window, Root, width, height);
+
+        runtime->Dispatch([](Babylon::Env& env)
+                          {
+                              Babylon::Console::CreateInstance(env, [](const char* message, Babylon::Console::LogLevel level)
+                              {
+                                  switch (level)
+                                  {
+                                      case Babylon::Console::LogLevel::Log:
+                                          __android_log_write(ANDROID_LOG_INFO, "BabylonNative", message);
+                                          break;
+                                      case Babylon::Console::LogLevel::Warn:
+                                          __android_log_write(ANDROID_LOG_WARN, "BabylonNative", message);
+                                          break;
+                                      case Babylon::Console::LogLevel::Error:
+                                          __android_log_write(ANDROID_LOG_ERROR, "BabylonNative", message);
+                                          break;
+                                  }
+                              });
+                          });
+
+        inputBuffer = std::make_unique<InputManager::InputBuffer>(*runtime);
+        InputManager::Initialize(*runtime, *inputBuffer);
+
+        runtime->LoadScript("Scripts/babylon.max.js");
+        runtime->LoadScript("Scripts/babylon.glTF2FileLoader.js");
+        //runtime->LoadScript("Scripts/experience.js");
+
+        runtime->UpdateSize(width, height);
```

Comment:
Why do we need this Update when we just passed the width/height to the constructor. Should it be doing this internally?

Reply:
Removed